web/src/screens/filters/action.tsx

```diff
@@ -106,14 +107,51 @@ interface TypeFormProps {
 }
 
 const TypeForm = ({ action, idx, clients }: TypeFormProps) => {
+  const { setFieldValue } = useFormikContext();
+  
+  const resetClientField = (action: Action, idx: number, prevActionType: string): void => {
+    const fieldName = `actions.${idx}.client_id`;
+
+    if (prevActionType !== action.type && (
+      action.type === "QBITTORRENT" ||
+      action.type === "DELUGE_V1" ||
+      action.type === "DELUGE_V2" ||
+      action.type === "RTORRENT" ||
+      action.type === "TRANSMISSION" ||
+      action.type === "PORLA" ||
+      action.type === "RADARR" ||
+      action.type === "SONARR" ||
+      action.type === "LIDARR" ||
+      action.type === "WHISPARR" ||
+      action.type === "READARR" ||
+      action.type === "SABNZBD"
+    )) {
+      setFieldValue(fieldName, ""); // Reset the client_id field value
+    }
+  };
+
+  const [prevActionType, setPrevActionType] = useState<string | null>(null);
+  
   switch (action.type) {
   case "TEST":
+    useEffect(() => {
+      if (prevActionType !== null) {
+        resetClientField(action, idx, prevActionType);
+      }
+      setPrevActionType(action.type);
+    }, [action.type, idx, setFieldValue]); 
```

Comment:
This can probably be unified below line `133` to not have to repeat for each case. I might be wrong, but that is before rendering.

Reply:
I actually tried something like this, but I think I placed it below L135 instead. Seems to work fine when putting it above the switch!